modules/react/collection/stories/examples/MultiSelection.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+import {
+  useListItemRegister,
+  useListItemRovingFocus,
+  useListRenderItems,
+  useListItemSelect,
+  useListModel,
+  multiSelectionManager,
+  ListProps,
+  ListItemProps,
+} from '@workday/canvas-kit-react/collection';
+import {composeHooks} from '@workday/canvas-kit-react/common';
+
+const ListModelContext = useListModel.Context;
+
+const List = (props: ListProps) => {
+  const model = useListModel({
+    selection: multiSelectionManager,
+  });
+
+  return (
+    <ListModelContext.Provider value={model}>
+      <ul>{useListRenderItems(model, props.children)}</ul>
+      <p>Cursor ID: {model.state.cursorId}</p>
+      <p>
+        Selected ID: {(model.state.selectedIds !== 'all' ? model.state.selectedIds : []).join(',')}
+      </p>
+    </ListModelContext.Provider>
+  );
+};
+
+const useItem = composeHooks(useListItemSelect, useListItemRovingFocus, useListItemRegister);
```

Comment:
the order of these matters right? is it `useListItemRegister` -> `useListItemRovingFocus` -> `useListItemSelect` where you need to register your items first and your select is last?

Reply:
This one is weird and I tried to change it when I created it.

It runs hooks from right to left, but merges props from left to right. So the rightmost runs first, while the leftmost props win.

For example, the lodash `compose` works the same way - `compose` runs functions from right to left.